tests/tests_fabric/plugins/precision/test_bitsandbytes.py

```diff
@@ -0,0 +1,143 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+import sys
+from unittest.mock import Mock
+
+import lightning.fabric
+import pytest
+import torch
+import torch.distributed
+from lightning.fabric import Fabric
+from lightning.fabric.connector import _Connector
+from lightning.fabric.plugins.precision.bitsandbytes import _BITSANDBYTES_AVAILABLE, BitsandbytesPrecision
+
+from tests_fabric.helpers.runif import RunIf
+
+
+@pytest.mark.skipif(_BITSANDBYTES_AVAILABLE, reason="bitsandbytes needs to be unavailable")
+def test_bitsandbytes_plugin(monkeypatch):
+    module = lightning.fabric.plugins.precision.bitsandbytes
+    monkeypatch.setattr(module, "_BITSANDBYTES_AVAILABLE", lambda: True)
+    bitsandbytes_mock = Mock()
+    monkeypatch.setitem(sys.modules, "bitsandbytes", bitsandbytes_mock)
+
+    class ModuleMock(torch.nn.Linear):
+        ...
+
+    class NF4LinearMock(ModuleMock):
+        def _quantize_weight(self, w):
+            pass
+
+    module._FP4Linear = ModuleMock
+    module._NF4Linear = NF4LinearMock
+    module._Int8LinearInference = ModuleMock
+    module._FP4DQLinear = ModuleMock
+    module._NF4DQLinear = ModuleMock
+    module._Linear4bit = ModuleMock
+    module._Linear8bitLt = ModuleMock
+
+    precision = BitsandbytesPrecision("nf4", dtype=torch.float16)
+    connector = _Connector(plugins=precision)
+    assert connector.precision is precision
+    assert precision.dtype == torch.float16
```

Comment:
could we move the connector-specific unit tests to test_connector?

Reply:
This is a bit annoying to do because there's like 30 lines of mocking setup which would now be dupe in two tests. Knowing this, do you prefer to split it?

Reply:
Fine with me if it's more practical this way. Could also think about a shared fixture that mocks if that makes it more managable.

Reply:
Maybe in the future if we add more tests